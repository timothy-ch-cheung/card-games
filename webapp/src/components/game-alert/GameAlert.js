import React from "react";
import Alert from "react-bootstrap/Alert";
import Fade from "react-bootstrap/cjs/Fade";

function GameAlert(props) {
    const alertStyle = {
        position: "fixed",
        zIndex: 999,
        textAlign: "left",
        top: "10%",
        left: 0,
        right: 0,
        marginLeft: "auto",
        marginRight: "auto",
        padding: "10px"
    }
    if (props.show) {
        return (
            <>
                <style type="text/css">
                    {`
                        .alert-dismissible .close {
                            padding: 5px 10px;
                        }
                        .alert-danger {
                            width: 60%;
                        }
                        @media(max-width: 800px) {
                            .alert-danger {
                                width: 90%;
                            }
                        }
                `}
                </style>
                <Alert variant="danger" style={alertStyle} onClose={() => props.onClose()} dismissible>
                    <Alert.Heading>Error!</Alert.Heading>
                    <p style={{textAlign: "center", marginBottom: "2px"}}>{props.errorText}</p>
                </Alert>
            </>
        )
    }
    return null;
}

export default GameAlert;