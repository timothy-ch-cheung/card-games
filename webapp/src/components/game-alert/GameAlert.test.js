import React from 'react';
import {shallow} from "enzyme";
import GameAlert from "./GameAlert";

describe("TEST SUITE Game Alert: ", () => {
    test('Matches snapshot', () => {
        let wrapper = shallow(<GameAlert show={true} errorText="Error: Game not found"/>)
        expect(wrapper).toMatchSnapshot()
    });
});