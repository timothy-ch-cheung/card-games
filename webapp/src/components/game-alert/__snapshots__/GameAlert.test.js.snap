// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST SUITE Game Alert:  Matches snapshot 1`] = `
<Fragment>
  <style
    type="text/css"
  >
    
                        .alert-dismissible .close {
                            padding: 5px 10px;
                        }
                        .alert-danger {
                            width: 60%;
                        }
                        @media(max-width: 800px) {
                            .alert-danger {
                                width: 90%;
                            }
                        }
                
  </style>
  <Alert
    closeLabel="Close alert"
    dismissible={true}
    onClose={[Function]}
    show={true}
    style={
      Object {
        "left": 0,
        "margin": "10px auto",
        "padding": "10px",
        "position": "fixed",
        "right": 0,
        "textAlign": "left",
        "top": "10%",
        "zIndex": 999,
      }
    }
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "displayName": "Fade",
        "render": [Function],
      }
    }
    variant="danger"
  >
    <AlertHeading>
      Error!
    </AlertHeading>
    <p
      style={
        Object {
          "marginBottom": "2px",
          "textAlign": "center",
        }
      }
    >
      Error: Game not found
    </p>
  </Alert>
</Fragment>
`;
